src/sdk/namespace_fs.js

```diff
@@ -1127,6 +1134,20 @@ class NamespaceFS {
             throw err;
         }
     }
+
+    async _validate_data_access(fs_account_config, file_path) {
```

Comment:
The name doesn't fit. I would make it explicit - _fail_if_archived_or_sparse_file()